vars/mavenExecuteStaticCodeChecks.groovy

```diff
@@ -1,11 +1,45 @@
+import com.sap.piper.ConfigurationLoader
 import com.sap.piper.DownloadCacheUtils
+import com.sap.piper.QualityCheck
+import com.sap.piper.ReportAggregator
 import groovy.transform.Field
 
+import static com.sap.piper.Prerequisites.checkScript
+
 @Field String METADATA_FILE = 'metadata/mavenStaticCodeChecks.yaml'
 @Field String STEP_NAME = getClass().getName()
 
 void call(Map parameters = [:]) {
+    final script = checkScript(this, parameters) ?: null
     List credentials = []
-    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(parameters.script, parameters)
-    piperExecuteBin(parameters, STEP_NAME, METADATA_FILE, credentials)
+    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(script, parameters)
+
+    try {
+        piperExecuteBin(parameters, STEP_NAME, METADATA_FILE, credentials)
+    } catch (Exception exception) {
+        error("Maven Static Code Checks execution failed. Please examine the reports which are also available in the Jenkins user interface.")
```

Comment:
Could it be helpful to log the exception here including stack-trace? Or do you think it would be in the log anyway as output from the go step?

Reply:
that can be viewed in the maven log and after that the user will wnd up locally or have a look into the reports.

Reply:
I think we do not need a stack trace.